### DISCUSSION TITLE
NFT QR code

### DISCUSSION THREAD

**Discussion by AnouarBF** [Cyfrin/foundry-full-course-cu#19] [Q&A] [answered]
I just started the course and fineshed lesson 0. I don't know how exactly to use these QR code that appears at the end of the lesson

**Comment by alymurtazamemon** [comment, marked as answer]
@AnouarBF You need to scan these QR codes using a QR code scanner if you are using an Android phone or iPhone camera by default detect these. Whatever you have, after scanning those will give you a URL (each lesson has 2 QR codes, 1 for sepolia which is testnet so you can play around to solve the challenge with free testnet ethers and 1 for arbitrum where the challenge will be the same just you will be needed real ether to pay for the gas).

Open the URL in the new tab and in the first contract code you will find the `solveChallenge` function where you will find challenges for each lesson. Once you find the answer, you can submit it to see if you succeed or not. 

NOTE: If metamask shows your error that it cannot estimate gas then it means you did not solve the challenge yet.

**Comment by cromewar** [comment]
@alymurtazamemon  is right, also remember that if you decide to actually pay to solve the challenge on arbitrum you will get cool NFTs.

**Comment by nik132-eng** [comment]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/77881638/201ccb84-c1ec-4582-89fb-736fd0d14946)


from where do we need to compile code and solve challenges? @cromewar @alymurtazamemon

**Reply by PatrickAlphaC** [reply]
If it's not clear to you, continue the course and come back to it :)

**Reply by alymurtazamemon** [reply]
@nik132-eng You do not need to compile it, just here right to the `Code` tab you will find the `Write Contract` tab, go there connect your wallet and just use the `solveChallenge` function (but make sure for each challenge you read the challenge description).
